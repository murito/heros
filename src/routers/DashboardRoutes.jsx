import React from 'react'
import { Navbar } from '../components/ui/Navbar'
import { Switch, Route, Redirect } from 'react-router-dom';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heros/HeroScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container-fluid mt-5 px-5">
                <Switch>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/search" component={ SearchScreen }/>

                    <Redirect to="/marvel" />
                </Switch>    
            </div>   
        </>
    )
}
