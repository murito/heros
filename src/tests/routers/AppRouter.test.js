import { shallow, mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Tests on <AppRouter />', () => {
    let contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }
    } 

    
    test('should show the login if the user is not logged', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the Marvel component if the user is logged', () => {
        contextValue = {
            ...contextValue,
            user: {
                logged: true
            }
        };

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBeTruthy();
    })
    
    
})
