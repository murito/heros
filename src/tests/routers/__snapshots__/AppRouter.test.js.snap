// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> should show the login if the user is not logged 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
